pyccel/parser/syntax/headers.py

```diff
@@ -495,11 +520,12 @@ def expr(self):
             else:
                 NotImplementedError("Unrecognised macro argument type")
 
-
-        results = self.results
-        if (results is None):
-            results = []
-
+        results = []
+        results_shapes = []
+        if self.results:
+            results = [res.expr['name'] for res in self.results]
+            results = [PyccelSymbol(r) for r in results]
+            results_shapes = [res.expr['shape'] for res in self.results]
```

Comment:
For clarity I recommend that we avoid redefining the same variable multiple times. And if we can make sure that `self.results` is always a list (even if empty - see my previous comment), this code reduces to just two lines:
```python
results = [PyccelSymbol(r.expr['name']) for r in self.results]
results_shapes = [r.expr['shape'] for r in self.results]
```

Reply:
Fixed 3d84be1